base_layer/wallet/src/ffi.rs

```diff
@@ -120,284 +435,574 @@ pub unsafe extern "C" fn create_pending_transaction_outputs(
+        return false;
+    }
+
+    (*wallet).output_manager_service.add_output((*output).clone()); // ? true : false; implement AddOutput(O: UnblindedOutput) -> bool on Wallet
+                                                                    // i.e return (*wallet).addOutput((*output));
+    return true;
+}
+
+/// Append an UnblindedOutput to be spent to the pending transaction outputs object
+#[no_mangle]
+pub unsafe extern "C" fn wallet_add_output_to_spend(
+    wallet: *mut TariWallet,
+    output: *mut WalletUnblindedOutput,
 ) -> bool
 {
-    // append input to tx
+    if wallet.is_null() {
+        return false;
+    }
+
+    if output.is_null() {
+        return false;
+    }
+
+    // (*wallet).pendingtransactionoutputs.addSpendoutput((*output)) ? true : false;
+    return true;
 }
 
-/// Add a transaction output to a transaction struct
+/// Append an UnblindedOutput to be received to the pending transaction outputs object
 #[no_mangle]
-pub unsafe extern "C" fn add_transaction_output(
-    transaction: *mut Transaction,
-    commitment: *const c_char, // Byte[32] - Commitment
-    proof: *const c_char,      // Byte[32] - Rangeproof
-    feature_flags: c_char,     // OutputFlags,
-    maturity: c_ulonglong,     // u64
+pub unsafe extern "C" fn wallet_add_output_to_received(
+    wallet: *mut TariWallet,
+    output: *mut WalletUnblindedOutput,
 ) -> bool
 {
-    // append output to tx
+    if wallet.is_null() {
+        return false;
+    }
+
+    if output.is_null() {
+        return false;
+    }
+
+    // (*wallet).pendingtransactionoutputs.addReceivedoutput((*output)) ? true : false;
+    return true;
 }
 
-/// Add a transaction kernel to a transaction struct
+/// Add an output to the wallet. `spent` is a boolean that indicates if this output is a spent or unspent output.
 #[no_mangle]
-pub unsafe extern "C" fn add_transaction_kernel(
-    transaction: *mut Transaction,
-    features: c_char, // KernelFeatures,
-    fee: c_ulonglong, // MicroTari,
-    lock_height: u64,
-    meta_info: *const c_char,     // Option<Byte[32]> - Option<HashOutput>,
-    linked_kernel: *const c_char, // Option<Byte[32]> - Option<HashOutput>,
-    excess: *const c_char,        // Byte[32] - Commitment,
-    excess_sig: *const c_char,    // Byte[32] - Signature,
+pub unsafe extern "C" fn wallet_add_pending_transaction_outputs(
+    wallet: *mut Wallet,
+    output: *mut PendingTransactionOutputs,
+    spent: bool,
 ) -> bool
 {
-    // append kernel to tx
+    if wallet.is_null() {
+        return false;
+    }
+
+    if output.is_null() {
+        return false;
+    }
+
+    match spent {
+        true => {},  //(*wallet).pendingtransactionoutputs.addSpentOutput((*output)) ? true : false;
+        false => {}, //(*wallet).pendingtransactionoutputs.addReceivedOutput((*output)) ? true : false;
+    }
+    return true;
+}
+
+/// TODO Methods to construct, free above 3 types
+
+#[no_mangle]
+pub unsafe extern "C" fn wallet_create_transaction(
```

Comment:
This should be next to the methods that add tx_inputs, tx_outputs and tx_kernels as they are related.